lib/Doctrine/ORM/Query/Expr/Composite.php

```diff
@@ -63,7 +66,8 @@ private function processQueryPart($part)
         }
 
         // Fixes DDC-1237: User may have added a where item containing nested expression (with "OR" or "AND")
-        if (stripos($queryPart, ' OR ') !== false || stripos($queryPart, ' AND ') !== false) {
+        $singleLineQueryPart = str_replace(PHP_EOL, ' ', $queryPart);
+        if (stripos($singleLineQueryPart, ' OR ') !== false || stripos($singleLineQueryPart, ' AND ') !== false) {
```

Comment:
Maybe we would be better off with one single `preg_replace` here, with `\s(OR|AND)\s`. That way tabs are handled, and line feeds are preserved, that's what I'm getting at.

Reply:
Yeah, but maybe `str_replace` is faster. I will try `preg_replace`.